### PR TITLE
5.1

### PR DESCRIPTION
**Pull request by jellespijker**
See https://github.com/Ultimaker/Cura/pull/12708 for all the PR's

Failing tests are unrelated to this PR

### CODE REVIEW

File: .github/workflows/conan-package.yml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            - 'tests/**'
+            - 'conanfile.py'
+            - 'conandata.yml'
+            - 'CMakeLists.txt'
+            - '.github/workflows/conan-package.yml'
+            - '.github/worflows/requirements-conan-package.txt'
+        branches:
+            - main
+            - 'CURA-*'
+            - '[1-9]+.[0-9]+'
```
Comment (casperlamboo) [inline .github/workflows/conan-package.yml:38]:
```suggestion
            - '[0-9]+.[0-9]+'
```

File: .github/workflows/conan-package.yml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        tags:
+            - '[1-9]+.[0-9]+.[0-9]+'
```
Comment (casperlamboo) [inline .github/workflows/conan-package.yml:40]:
```suggestion
            - '[0-9]+.[0-9]+.[0-9]+'
```

File: .github/workflows/unit-test.yml [hunk 3]
```diff
@@ -8,74 +9,110 @@ on:
...
             - 'conanfile.py'
+            - 'conandata.yml'
             - 'CMakeLists.txt'
             - '.github/workflows/unit-test.yml'
+            - '.github/worflows/requirements-conan-package.txt'
         branches:
             - main
             - 'CURA-*'
-            - 'CT_fix'
+            - '[1-9]+.[0-9]+'
```
Comment (casperlamboo) [inline .github/workflows/unit-test.yml:19]:
```suggestion
            - '[0-9]+.[0-9]+'
```

File: .github/workflows/unit-test.yml [hunk 4]
```diff
@@ -8,74 +9,110 @@ on:
...
         tags:
-            - '[5-9].[0-9].+*'
+            - '[1-9]+.[0-9]+.[0-9]+'
```
Comment (casperlamboo) [inline .github/workflows/unit-test.yml:21]:
```suggestion
            - '[0-9]+.[0-9]+.[0-9]+'
```

File: .github/workflows/unit-test.yml [hunk 5]
```diff
@@ -8,74 +9,110 @@ on:
...
             - 'conanfile.py'
+            - 'conandata.yml'
             - 'CMakeLists.txt'
             - '.github/workflows/unit-test.yml'
+            - '.github/worflows/requirements-conan-package.txt'
         branches:
             - main
-            - '[5-9].[0-9]'
+            - 'CURA-*'
+            - '[1-9]+.[0-9]+'
```
Comment (casperlamboo) [inline .github/workflows/unit-test.yml:35]:
```suggestion
            - '[0-9]+.[0-9]+'
```

File: .github/workflows/unit-test.yml [hunk 6]
```diff
@@ -8,74 +9,110 @@ on:
...
+        tags:
+            - '[1-9]+.[0-9]+.[0-9]+'
```
Comment (casperlamboo) [inline .github/workflows/unit-test.yml:37]:
```suggestion
            - '[0-9]+.[0-9]+.[0-9]+'
```

**Review by casperlamboo — APPROVED** [review]

File: .github/workflows/conan-package.yml [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        branches:
+            - main
+            - 'CURA-*'
+            - '[1-9]+.[0-9]+'
+        tags:
+            - '[0-9]+.[0-9]+.[0-9]+'
+
+jobs:
+    conan-recipe-version:
+        uses: ultimaker/cura/.github/workflows/conan-recipe-version.yml@CURA-9365
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:44]:
```suggestion
        uses: ultimaker/cura/.github/workflows/conan-recipe-version.yml@main
```

File: .github/workflows/conan-package.yml [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+        with:
+            project_name: curaengine
+
+    conan-package-export:
+        needs: [ conan-recipe-version ]
+        uses: ultimaker/cura/.github/workflows/conan-recipe-export.yml@5.1
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:50]:
```suggestion
        uses: ultimaker/cura/.github/workflows/conan-recipe-export.yml@main
```

File: .github/workflows/conan-package.yml [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+            conan_config_branch: 'master'
+            conan_logging_level: 'info'
+        secrets: inherit
+
+    conan-package-create-macos:
+        # FIXME: For release branches: maybe rename the branch to release/**
+        if: ${{ (github.event_name == 'push' && (github.ref_name == 'main' || github.ref_name == '5.1' || github.ref_name == 'CURA-9365_fix_building_cura_main' )) || (github.event_name == 'workflow_dispatch' && inputs.create_binaries_macos) }}
+        needs: [ conan-recipe-version, conan-package-export ]
+
+        uses: ultimaker/cura/.github/workflows/conan-package-create.yml@5.1
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:66]:
```suggestion
        uses: ultimaker/cura/.github/workflows/conan-package-create.yml@main
```

File: .github/workflows/conan-package.yml [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+            conan_config_branch: 'master'
+            conan_logging_level: 'info'
+        secrets: inherit
+
+    conan-package-create-windows:
+        # FIXME: For release branches: maybe rename the branch to release/**
+        if: ${{ (github.event_name == 'push' && (github.ref_name == 'main' || github.ref_name == '5.1')) || (github.event_name == 'workflow_dispatch' && inputs.create_binaries_windows) }}
+        needs: [ conan-recipe-version, conan-package-export ]
+
+        uses: ultimaker/cura/.github/workflows/conan-package-create.yml@5.1
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:80]:
```suggestion
        uses: ultimaker/cura/.github/workflows/conan-package-create.yml@main
```

File: .github/workflows/conan-package.yml [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+            recipe_id_full: ${{ needs.conan-recipe-version.outputs.recipe_id_full }}
+            runs_on: 'windows-2022'
+            python_version: '3.10.4'
+            conan_config_branch: 'master'
+            conan_logging_level: 'info'
+        secrets: inherit
+
+    conan-package-create-linux:
+        # FIXME: For release branches: maybe rename the branch to release/**
+        if: ${{ (github.event_name == 'push' && (github.ref_name == 'main' || github.ref_name == '5.1')) || (github.event_name == 'workflow_dispatch' && inputs.create_binaries_linux) }}
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:91]:
```suggestion
        if: ${{ (github.event_name == 'push' && github.ref_name == 'main') || (github.event_name == 'workflow_dispatch' && inputs.create_binaries_linux) }}
```

File: .github/workflows/conan-package.yml [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+            success: ${{ contains(join(needs.*.result, ','), 'success') }}
+            success_title: "New Conan recipe exported in ${{ github.repository }}"
+            success_body: "Exported ${{ needs.conan-recipe-version.outputs.recipe_id_full }}"
+            failure_title: "Failed to export Conan Export in ${{ github.repository }}"
+            failure_body: "Failed to exported ${{ needs.conan-recipe-version.outputs.recipe_id_full }}"
+        secrets: inherit
+
+    notify-create:
+        # FIXME: For release branches: maybe rename the branch to release/**
+        if: ${{ always() && ((github.event_name == 'push' && (github.ref_name == 'main' || github.ref_name == '5.1')) || (github.event_name == 'workflow_dispatch' && inputs.create_binaries_linux)) }}
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:118]:
```suggestion
        if: ${{ always() && ((github.event_name == 'push' && github.ref_name == 'main') || (github.event_name == 'workflow_dispatch' && inputs.create_binaries_linux)) }}
```

File: .github/workflows/conan-package.yml [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+        needs: [ conan-package-create-macos, conan-package-create-windows, conan-package-create-linux ]
+
+        uses: ultimaker/cura/.github/workflows/notify.yml@5.1
```
Comment (Joeydelarago) [inline .github/workflows/conan-package.yml:121]:
```suggestion
        uses: ultimaker/cura/.github/workflows/notify.yml@main
```

**Review by Joeydelarago — COMMENTED** [review]
Update branch names

**Pull request merged**